src/apps/silo-finance/common/silo-finance.silo.contract-position-fetcher.ts

```diff
@@ -0,0 +1,233 @@
+import { Inject } from '@nestjs/common';
+import { BigNumber, BigNumberish } from 'ethers';
+import { formatEther } from 'ethers/lib/utils';
+import { gql } from 'graphql-request';
+
+import { APP_TOOLKIT, IAppToolkit } from '~app-toolkit/app-toolkit.interface';
+import { buildPercentageDisplayItem } from '~app-toolkit/helpers/presentation/display-item.present';
+import { getLabelFromToken } from '~app-toolkit/helpers/presentation/image.present';
+import { gqlFetch } from '~app-toolkit/helpers/the-graph.helper';
+import { DefaultDataProps, StatsItem } from '~position/display.interface';
+import { MetaType } from '~position/position.interface';
+import { ContractPositionTemplatePositionFetcher } from '~position/template/contract-position.template.position-fetcher';
+import {
+  GetDefinitionsParams,
+  DefaultContractPositionDefinition,
+  GetTokenDefinitionsParams,
+  UnderlyingTokenDefinition,
+  GetDisplayPropsParams,
+  GetTokenBalancesParams,
+  GetDataPropsParams,
+} from '~position/template/contract-position.template.types';
+
+import { Silo, SiloFinanceContractFactory, SiloLens } from '../contracts';
+import { IBaseSilo } from '../contracts/ethers/Silo';
+
+const GETSILOS_QUERY = gql`
+  query GetSilos {
+    markets(orderBy: totalValueLockedUSD, orderDirection: desc, where: { inputToken_: { activeOracle_not: "null" } }) {
+      id
+      name
+      isActive
+      inputToken {
+        symbol
+      }
+      outputToken {
+        symbol
+      }
+      __typename
+    }
+  }
+`;
+
+type SiloContractPositionDefinition = DefaultContractPositionDefinition & {
+  assets: string[];
+  tokens: string[][];
+  tokensFlattened: string[];
+  assetStorages: IBaseSilo.AssetStorageStructOutput[];
+};
+
+type SiloContractPositionDataProps = DefaultDataProps & {
+  totalUnderlyings: BigNumberish[];
+  totalShares: BigNumberish[];
+  supplyApys: BigNumberish[];
+  borrowApys: BigNumberish[];
+  tokensFlattened: string[];
+};
+
+export abstract class SiloFinanceSiloContractPositionFetcher extends ContractPositionTemplatePositionFetcher<
```

Comment:
Since it's a token it should be an app token instead of a contract position

Reply:
It needs to be contract position to better reason about the borrowing position